### ISSUE TITLE
Issues with number of voters in poll

### ISSUE THREAD

**Issue by anarute** [loomio/loomio#9067]
We found 2 issues in how the total of voters in a poll is displayed/counted:

1 - When creating a poll that is for everyone of the group to vote, the total of voters is the same of the people who voted, instead of being everyone in the group.

For example: I've created this poll bellow in a group that has 2 members. When creating the poll I chose "Anyone in the group", so I expected I didn't need to invite them in order for them to appear as voters. This creates a poll with 0 voters:

<img width="832" alt="Screen Shot 2022-10-12 at 14 02 58" src="https://user-images.githubusercontent.com/333447/195338239-7985f258-0864-4bce-b289-2800f32f5ca0.png">

2 - To solve this we need to invite a group to the poll anyway and then we face a new issue: everyone in the group is added except the person inviting, so the total is always missing 1 person until we manually invite this user. To give a better example: when I invite a group to a poll I have to invite the group + myself.

For the poll bellow, I've invited the whole group and as you can see my user wasn't added:

<img width="779" alt="Screen Shot 2022-10-12 at 14 06 59" src="https://user-images.githubusercontent.com/333447/195338587-8ed055f0-3f39-456b-adbc-82cb4bc000ce.png">

@robguthrie This is a critical issue for us because we use these numbers to officially validate a poll in our internal decision making process.

**Comment by robguthrie** [comment]
Hi @anarute. Thanks for reporting this. 

The number of voters in a poll is the number of votes created. When you invite someone to a poll, you create an uncast vote for them. So it's a simple count of the votes, cast or not cast yet.

It gets super complex if we try to count the number of people in the group, because the group size can change during and after the poll.

Previously to 2.15.0 - if you created a poll you always got a vote in it, which isn't always what people want.  So in v2.15.0 we made a change so that the author of the poll is not automatically added a poll they create. We intended to ensure that the author is included when they invite "everyone in the group", but there was a bug with that. So I've fixed that now.

In summary:
Is it OK for your needs to use "Anyone in the group" can vote, AND invite the group - if I fix the bug so that the author/current user is included when you do?

**Comment by robguthrie** [comment]
@anarute a fix for "inviting group does not invite author" is in the master branch now.

**Comment by anarute** [comment]
> It gets super complex if we try to count the number of people in the group, because the group size can change during and after the poll.

I see what you mean now. I think it's a matter of expectations then, when we create it as "anyone in the group" we expect all of them are invited and not that we also need to invite the whole group to make sure they are counted in the totals, but understanding now how it works I think we can change our workflow.

Thank you for fixing the author invitation issue. Btw I haven't tested yet but just to make sure that is the author of the invitation that is not added and not the author of the poll.

**Comment by robguthrie** [comment]
I'd like to get rid of the option entirely. I might be able to infer it when people invite the whole group to the poll.

That's right. It's actually the "actor" or person performing the action who was being filtered out. 
It's the right thing for most notifications, you often want to notify the whole group, but don't expect to notify yourself - the poll reminder is a good example. It's a bit different for a poll invitation, because now we don't add the actor automatically.
